billit/client.py

```diff
@@ -0,0 +1,68 @@
+import requests
+
+from .auth import BillitAuthentication
+from .constants import (
+    PRODUCTION_BASE_URL,
+    PRODUCTION_ENVIRONMENT,
+    SANDBOX_BASE_URL,
+    SANDBOX_ENVIRONMENT,
+)
+from .error import ApiError, AuthenticationError
+
+
+class Client:
+    """
+    Base client for all API clients
+    """
+
+    def __init__(self, api_key, environment=PRODUCTION_ENVIRONMENT):
+        self.api_key = api_key
+        self.account = Account(self)
+
+        if environment not in [PRODUCTION_ENVIRONMENT, SANDBOX_ENVIRONMENT]:
+            raise Exception(f"Invalid environment provided: {environment}")
+
+        if environment == PRODUCTION_ENVIRONMENT:
+            self.base_url = PRODUCTION_BASE_URL
+        else:
+            self.base_url = SANDBOX_BASE_URL
+
+    def _handle_response(self, response):
+        if response.status_code == 401:
+            raise AuthenticationError(response.json()["message"], response.status_code)
+
+        try:
+            response.raise_for_status()
+
+            return response.json()
+        except:
+            if "application/json" in response.headers["Content-Type"]:
+                error = f"{response.json()['message']}"
+            else:
+                error = f"{response.text}"
```

Comment:
No need for `if/else`. Let's just default:

```suggestion
            error = f"{response.text}"
            if "application/json" in response.headers["Content-Type"]:
                error = f"{response.json()['message']}"
```